Issue 658

adamrp opened the pull request.

Commit:
Merge branch 'elevation2' of https://github.com/adamrp/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'fix_study_summary' of https://github.com/adamrp/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'fixconsentformat' of https://github.com/teravest/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Add stored procedure to pull down animal metadata

Commit:
Add function to ag_data_access for new procedure

Commit:
Update script, pull down human AND animal barcodes

Fixes #658

Commit:
Update script, pull down human AND animal barcodes

Commit:
Merge branch 'master' of git://github.com/qiime/qiime_web_app

Commit:
Merge branch 'issue_658' of https://github.com/adamrp/qiime_web_app into issue_658

adamrp commented:
Works in preliminary tests, still needs more testing though!

ElDeveloper commented on python_code/scripts/ag_get_metadata.py:
```diff
@@ -38,14 +40,14 @@
 script_info['output_description']= (
-    "Results are written to an output file, one line per barcode. Column "
+    "Results are written to two output files, one file for humans, and one "
+    "file for animals.  In each file, there is one line per barcode. Column "
```
You have two spaces after `animals.`.

adamrp replied:
I always put two spaces after a concluding period! Perhaps a bit archaic, but it used to be the standard (and is still "acceptable" according to the [Modern Language Association's FAQ about it](http://www.mla.org/style/style_faq/mlastyle_spaces)).  [This Slate article](http://www.slate.com/articles/technology/technology/2011/01/space_invaders.html)'s author is pretty adamant about a single space, but then in one paragraph states that two spaces "was" used reasonably with monospaced fonts (as if monospaced fonts aren't used any more and "went out in the 1970s!").  Oh, great, and now I see that when github renders markdown, it removes double-spaces after periods!  Unbelievable. ![](http://www.reactiongifs.com/r/wwyp.gif)

adamrp replied:
I guess I'm not that consistent even in code, though, considering the other periods around that only have one space.  Okay okay, I'll remove it.

ElDeveloper commented on python_code/scripts/ag_get_metadata.py:
```diff
@@ -38,14 +40,14 @@
-    make_option('-o', '--output_file', type='new_filepath',
-                help="The output file, to which metadata will be written")
-
+    options_lookup['output_dir']
```
niiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiiice. 

This is a rather handy thing that commonly goes unused.

adamrp replied:
I love this utility!

ElDeveloper commented on python_code/scripts/ag_get_metadata.py:
```diff
@@ -74,21 +76,43 @@ def get_ag_metadata_bulk(barcodes):
-            yield metadata[0]
+            yield True, metadata[0]
+
+def get_ag_metadata_bulk_animal(barcodes):
```
Seems like this is replicating the code in line 65, do you think it might make sense to have an argument to decide which type of metadata it is that you want to retrieve?

adamrp replied:
Yeah, makes sense, I'll do it!

ElDeveloper commented on python_code/scripts/ag_get_metadata.py:
```diff
@@ -145,14 +183,42 @@ def main():
-    with open(output_fp, 'w') as out_file:
+    fails = []
+    human_output_fp = join(output_dir, 'human_md.txt')
+    with open(human_output_fp, 'w') as out_file:
```
You gotta love context managers :godmode:

adamrp replied:
I do!  Sometimes the extra indentation is annoying, but I think it's generally worth it :+1:

ElDeveloper commented on python_code/scripts/ag_get_metadata.py:
```diff
@@ -145,14 +183,42 @@ def main():
         all_barcodes.extend(barcodes.split(','))
 
-    with open(output_fp, 'w') as out_file:
+    fails = []
```
Maybe worth renaming to `human_failures` (I know, that sounds funny).

adamrp replied:
Hahahaha that does sound funny, but I think that's accurate, I'll change it.

ElDeveloper commented on python_code/scripts/ag_get_metadata.py:
```diff
@@ -145,14 +183,42 @@ def main():
+                # in this case (on fail), "metadata" will just be the barcode
+                fails.append(metadata)
+
+    fails2 = []
```
`animal_failures`?

adamrp replied:
Sure.  Might call it `double_failures` or something (these are barcodes that failed to hit the human AND animal lists)

ElDeveloper commented on python_code/scripts/ag_get_metadata.py:
```diff
@@ -145,14 +183,42 @@ def main():
+                # in this case (on fail), "metadata" will just be the barcode
+                fails2.append(metadata)
+
+    if fails2:
```
Any reason why this is not done for `fails`?

adamrp replied:
`fails` is what failed to hit the human list. It only checks the animal list for the things that failed to hit the human list (see line 208).  If it fails _again_ when checking the animal list, then it truly is a failure.

Commit:
Address comments from @ElDeveloper

Commit:
Merge branch 'issue_658' of https://github.com/adamrp/qiime_web_app into issue_658

Conflicts:
	python_code/scripts/ag_get_metadata.py

Commit:
typos

adamrp commented:
I did some editing on webdev (much easier to test from there) and there are for some reason a bunch of old commits included now.  Still only 3 files changed, so I don't know what's going on there.  Anyways, I addressed the comments from @ElDeveloper and I did some further testing; it works, so should be ready for merge unless there are further comments.

adamrp commented:
Actually NOT READY FOR MERGE, because I didn't include the multiples in the stored procedure.

Commit:
Add multiples

Commit:
Merge branch 'issue_658' of github.com:adamrp/qiime_web_app into issue_658

Commit:
Add multiples to script

adamrp commented:
Okay, now this is ready for merge :)

ElDeveloper commented:
Nice, thanks for making those changes @adamrp. @teravest can you take a look and if you are ok with this would you mind merging it?
